functions/go/create-setters/createsetters/create_setters.go

```diff
@@ -0,0 +1,275 @@
+package createsetters
+
+import (
+	"fmt"
+	"strings"
+
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+var _ kio.Filter = &CreateSetters{}
+
+// CreateSetters creates a comment for the resource fields which 
+// contains the same value as setter reference value 
+type CreateSetters struct {
+	// Setters holds the user provided values for simple map setters
+	Setters []Setter
+
+	// Setters holds the user provided values for array setters
+	ArraySetters []ArraySetter
+
+	// Results are the results of applying setter values
+	Results []*Result
+
+	// filePath file path of resource
+	filePath string
+}
+
+// Setter stores name and value of the map setter
+type Setter struct {
+	// Name is the name of the setter
+	Name string
+
+	// Value is the input value for setter
+	Value string
+}
+
+// ArraySetter stores name and values of the array setter
+type ArraySetter struct {
+	// Name is the name of the setter
+	Name string
+
+	// ValueSet is the set of the values for setter
+	ValueSet map[string]bool
```

Comment:
Why is a map ? It can just be an array []string. Rename it to `Values`

Reply:
Later we need to check if all the values of array-setter and node are equal (referring to `checkEqual` function). I think it would reduce the complexity if it is a set.

Reply:
duplicate values are allowed in yaml sequence. You will loose them if you use set/map. You should achieve correctness first and then optimize if needed. Readability is also important.

On a side different note, you can use sets library instead of map if you want to use set.

Reply:
Good point. I didn't consider duplicate values.